cmd/mailchain/internal/http/server.go

```diff
@@ -65,9 +65,9 @@ func CreateRouter(s *settings.Root, cmd *cobra.Command) (http.Handler, error) {
 
 	api.HandleFunc("/envelope", handlers.GetEnvelope(config.sentStore)).Methods("GET")
 
-	api.HandleFunc("/messages", handlers.GetMessages(config.mailboxStateStore, config.cache, config.receivers, config.keystore, deriveKeyOptions)).Methods("GET") //nolint:lll
+	api.HandleFunc("/messages", handlers.GetMessages(config.mailboxStateStore, config.receivers, config.keystore, deriveKeyOptions)).Methods("GET") //nolint:lll
 	api.HandleFunc("/messages", handlers.SendMessage(config.sentStore, config.senders, config.keystore, deriveKeyOptions)).Methods("POST")
-
+	api.HandleFunc("/messages/fetch", handlers.FetchMessages(config.mailboxStateStore, config.cache, config.receivers, config.keystore, deriveKeyOptions)).Methods("POST")
```

Comment:
Ahhh yeah good spot, POST against `/messages` will not work since its used. Your solution looks good for this.